service/src/main/java/com/codeforcommunity/processor/ProtectedSiteProcessorImpl.java

```diff
@@ -876,6 +877,24 @@ && isAlreadyAdopted(siteId)
   }
 
   @Override
+
+  public List<SiteEntryImage> getUnapprovedImages(JWTData userData) {
+    assertAdminOrSuperAdmin(userData.getPrivilegeLevel());
+    List<SiteImagesRecord> imageRecords =
+            db.selectFrom(SITE_IMAGES).where(
+                    SITE_IMAGES.APPROVAL_STATUS.eq(ImageApprovalStatus.SUBMITTED.getApprovalStatus())).fetch();
+    List<SiteEntryImage> unapprovedImages = imageRecords.stream().map(
+            imageRecord ->
+                    new SiteEntryImage(
```

Comment:
Is it possible to pull out the SiteEntryImage creation, or the Users SQL query, into a helper method? This makes sense, but is a bit nested to read

Reply:
extracted